release/cloudbuild_presubmit.yaml

```diff
@@ -0,0 +1,91 @@
+steps:
+  # First create a fake tag we'll use throughout the CI build process below.
+  # Unfortunately, GCB has no concept of dynamically creating substitutions or
+  # passing ENV vars between steps, so the best we can do is to create a file
+  # containing our tag in the share workspace which other steps can inspect.
+  - name: bash
+    script: |
+      date +'0.2.%s-incompatible' > /workspace/fake_tag
+      cat /workspace/fake_tag
+  ### Build the Trusted Applet and upload it to GCS.
+  # Build an image containing the trusted applet artifacts with the Dockerfile.
+  # This step needs to be a bash script in order to substitute fake tag into a
+  # build arg.
+  - name: gcr.io/cloud-builders/docker
+    entrypoint: bash
+    args:
+      - -c 
+      - |
+        docker build \
+        --build-arg=TAMAGO_VERSION=${_TAMAGO_VERSION} \
+        --build-arg=FT_LOG_URL=${_LOG_BASE_URL} \
+        --build-arg=FT_BIN_URL=${_BIN_BASE_URL} \
+        --build-arg=LOG_ORIGIN=${_ORIGIN} \
+        --build-arg=LOG_PUBLIC_KEY=${_LOG_PUBLIC_KEY} \
+        --build-arg=APPLET_PUBLIC_KEY=${_APPLET_PUBLIC_KEY} \
+        --build-arg=OS_PUBLIC_KEY1=${_OS_PUBLIC_KEY1} \
+        --build-arg=OS_PUBLIC_KEY2=${_OS_PUBLIC_KEY2} \
+        --build-arg=GIT_SEMVER_TAG=$(cat /workspace/fake_tag) \
+        -t builder-image \
+        .
+  # Prepare a container with a copy of the artifacts.
+  - name: gcr.io/cloud-builders/docker
+    args:
+      - create
+      - --name
+      - builder_scratch
+      - builder-image
+  # Copy the artifacts from the container to the Cloud Build VM.
+  - name: gcr.io/cloud-builders/docker
+    args:
+      - cp
+      - builder_scratch:/build/bin
+      - output
+  # List the artifacts.
+  - name: bash
+    args:
+      - ls
+      - output
+  ### Construct log entry / Claimant Model statement.
+  # This step needs to be a bash script in order to substitute the fake take
+  # in the command args.
+  - name: golang
+    entrypoint: bash
+    args:
+      - -c
+      - |
+        go run github.com/transparency-dev/armored-witness/cmd/manifest@${_ARMORED_WITNESS_REPO_VERSION} \
+        create \
+        --git_tag=$(cat /workspace/fake_tag) \
+        --git_commit_fingerprint=${COMMIT_SHA} \
+        --firmware_file=output/trusted_applet.elf \
+        --firmware_type=TRUSTED_APPLET \
+        --tamago_version=${_TAMAGO_VERSION} \
+        --raw \
+        --output_file=output/trusted_applet_manifest_unsigned.json
+  # Sign the log entry.
+  - name: golang
+    args:
+      - go
+      - run
+      - github.com/transparency-dev/armored-witness/cmd/sign@${_ARMORED_WITNESS_REPO_VERSION}
```

Comment:
What key is this using to sign it with?

Reply:
The `sign` cmd knows how to map the `--release=ci` and `--artefact=applet` to the CI applet KMS key. I guess you are asking about a dedicated presubmit key? Ideally we would do that, I didn't think it was worth the work (since the keys are not GA yet, we need to ask the KMS team to help us create it in our project manually), I can do that if we think it's important. WDYT?

Reply:
I feel uneasy about reusing the CI keys for presubmit, but not enough to say that a bunch of extra work for allocating keys is definitely required. I would settle for this going in with a clear PR description that this uses the CI keys and an issue being raised so we don't lose sight of this. We have talked about making CI releases be automatically pushed out to some test devices, and when that happens we may want to close the backdoor to CI-signed releases being made that were not actually merged into `main`.

Reply:
I think running the `sign` tool here is probably only marginal value over checking that the much more complex previous steps are working as intended, so we could remove this signing step and avoid the issue?

Reply:
> I think running the `sign` tool here is probably only marginal value over checking that the much more complex previous steps are working as intended, so we could remove this signing step and avoid the issue?

Done, just pushed.